### PR TITLE
HCAP-1442 - Add site type to sites form/view

### PR DESCRIPTION
**Pull request by mmumby**
## Work completed
- Added 'siteType' to employer_sites create and edit form
- Displayed new value in the sites table
- Displayed new value in the site detail page
- Updated test suits with new required field

UI - UI title is now "Site Type"
Screen Shot 2023-03-28 at 10 20 46 AM
Screen Shot 2023-03-28 at 10 19 36 AM

Screen Shot 2023-03-28 at 10 21 07 AM
Screen Shot 2023-03-28 at 10 21 27 AM
Screen Shot 2023-03-28 at 10 23 12 AM

### CODE REVIEW

File: server/validators/employer.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ export const EmployerFormSchema = yup
       .test('is-true', errorMessage, (v) => v === true),
   });
 
+// TODO: determine if this is needed
 export const EmployerSiteBatchSchema = yup.array().of(
   yup.lazy((item, options) => {
```
Comment (mmumby) [inline server/validators/employer.ts:158]: This doesn't appear to be used anywhere - Didn't want to remove as part of this PR, but wanted to raise it as tech debt.

File: client/src/components/modal-forms/EditSiteForm.js [hunk 2]
```diff
@@ -51,6 +52,7 @@ export const EditSiteForm = ({
             />
             <Field name='address' component={RenderTextField} label='* Street Address' />
             <Field name='city' component={RenderTextField} label='* City' />
+            <Field name='postalCode' component={RenderTextField} label='* Postal Code' />
```
Comment (fw-noel) [inline client/src/components/modal-forms/EditSiteForm.js:55]: How come the reordering? I would put site type under postal code, I don't think it looks great under the radio

**Review by fw-noel — COMMENTED** [review]
If I edit a site and try to submit without a site type, it's still submitting it with a bad request response. Not being caught by the validator

**Comment by fw-noel** [comment]
I'm also noticing there's some different required fields for Edit vs. New site, I'll have to ask about that

File: client/src/components/modal-forms/EditSiteForm.js (hunk 2 above)
Comment (mmumby) [reply client/src/components/modal-forms/EditSiteForm.js:55]: I did this as it seemed like postalCode should go with address/city but was being broken up by the is HA radio buttons. 

Show i move postal code back below isHA? or just move siteType above isHA and keep postalCode under city?

**Comment by mmumby** [comment]
@fw-noel It appears the create form allows the minimum amount of data to be collected, and the edit form forces users to add more information.

Added the required to edit form, let me know what order is preferred. This is current state: 
Screen Shot 2023-03-28 at 3 17 07 PM

**Review by fw-noel — APPROVED** [review]
lgtm

**Pull request merged**